discovery/syncer_test.go

```diff
@@ -783,52 +783,52 @@ func TestGossipSyncerReplyChanRangeQuery(t *testing.T) {
 				t.Fatalf("expected ReplyChannelRange instead got %T", msg)
 			}
 
-			// Only for the first iteration do we set the offset to
-			// zero as no chunks have been processed yet. For every
-			// other iteration, we want to move forward by the
-			// chunkSize (from the staring block height).
-			offset := 0
-			if i != 0 {
-				offset = 1
-			}
-			expectedFirstBlockHeight := (i+offset)*2 + startingBlockHeight
+			// We'll determine the correct values of each field in
+			// each response based on the order that they were sent.
+			var expectedFirstBlockHeight, expectedNumBlocks uint32
+			var expectedComplete uint8
 
 			switch {
-			// If this is not the last chunk, then Complete should
-			// be set to zero. Otherwise, it should be one.
-			case i < 2 && rangeResp.Complete != 0:
-				t.Fatalf("non-final chunk should have "+
-					"Complete=0: %v", spew.Sdump(rangeResp))
-
-			case i < 2 && rangeResp.NumBlocks != chunkSize+1:
-				t.Fatalf("NumBlocks fields in resp "+
-					"incorrect: expected %v got %v",
-					chunkSize+1, rangeResp.NumBlocks)
-
-			case i < 2 && rangeResp.FirstBlockHeight !=
-				uint32(expectedFirstBlockHeight):
-
-				t.Fatalf("FirstBlockHeight incorrect: "+
-					"expected %v got %v",
-					rangeResp.FirstBlockHeight,
-					expectedFirstBlockHeight)
-			case i == 2 && rangeResp.Complete != 1:
-				t.Fatalf("final chunk should have "+
-					"Complete=1: %v", spew.Sdump(rangeResp))
-
-			case i == 2 && rangeResp.NumBlocks != 1:
-				t.Fatalf("NumBlocks fields in resp "+
-					"incorrect: expected %v got %v", 1,
-					rangeResp.NumBlocks)
-
-			case i == 2 && rangeResp.FirstBlockHeight !=
-				queryResp[len(queryResp)-1].BlockHeight:
-
-				t.Fatalf("FirstBlockHeight incorrect: "+
-					"expected %v got %v",
-					rangeResp.FirstBlockHeight,
-					queryResp[len(queryResp)-1].BlockHeight)
+			// The first reply should range from our starting block
+			// height until it reaches its maximum capacity of
+			// channels.
+			case i == 0:
```

Comment:
Great simplification of the prior `switch` statement, I had an itch to massage it a bit when I touched this area last, but ended up deferring it.